builtin/credential/radius/path_config.go

```diff
@@ -0,0 +1,224 @@
+package radius
+
+import (
+	"strings"
+
+	"github.com/fatih/structs"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func pathConfig(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: `config`,
+		Fields: map[string]*framework.FieldSchema{
+			"host": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Default:     "127.0.0.1",
+				Description: "radius host to connect to (default: 127.0.0.1)",
+			},
+
+			"port": &framework.FieldSchema{
+				Type:        framework.TypeInt,
+				Default:     1812,
+				Description: "radius port (default: 1812)",
+			},
+			"secret": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "secret shared with the radius server",
+			},
+			"allow_unknown_users": &framework.FieldSchema{
+				Type:        framework.TypeBool,
+				Default:     false,
+				Description: "enable granting default policy upon successful RADIUS authentication (default: false)",
+			},
+			"reauth_on_renew": &framework.FieldSchema{
+				Type:        framework.TypeBool,
+				Default:     false,
+				Description: "attempt reauthentication with backend before granting token renewal (default: false)",
+			},
+			"dial_timeout": &framework.FieldSchema{
+				Type:        framework.TypeDurationSecond,
+				Default:     10,
+				Description: "number of seconds before connect timeouts (default: 10)",
+			},
+			"read_timeout": &framework.FieldSchema{
+				Type:        framework.TypeDurationSecond,
+				Default:     10,
+				Description: "number of seconds before response timeouts (default: 10)",
+			},
+			"nas_port": &framework.FieldSchema{
+				Type:        framework.TypeInt,
+				Default:     10,
+				Description: "RADIUS NAS port field (default: 10)",
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation:   b.pathConfigRead,
+			logical.UpdateOperation: b.pathConfigWrite,
+		},
+
+		HelpSynopsis:    pathConfigHelpSyn,
+		HelpDescription: pathConfigHelpDesc,
+	}
+}
+
+/*
+ * Construct ConfigEntry struct using stored configuration.
+ */
+func (b *backend) Config(req *logical.Request) (*ConfigEntry, error) {
+	// Schema for ConfigEntry
+	fd, err := b.getConfigFieldData()
```

Comment:
`pathConfigRead` already is getting the FieldData as an argument. This function can take in that and fetch the required parameters. `getConfigFieldData` may not be required.